x/concentrated-liquidity/position_test.go

```diff
@@ -0,0 +1,121 @@
+		name              string
+		param             param
+		positionExists    bool
+		expectedLiquidity sdk.Dec
+		expectedErr       error
+	}{
+		{
+			name: "Init position from -50 to 50 with 50000000000 liquidity",
+			param: param{
+				poolId:         validPoolId,
+				lowerTick:      -50,
+				upperTick:      50,
+				liquidityDelta: DefaultLiquidityAmt,
+			},
+			positionExists:    false,
+			expectedLiquidity: DefaultLiquidityAmt,
+		},
+		{
+			name: "Update position from -50 to 50 that already contains 50000000000 liquidity with 50000000000 more liquidity",
+			param: param{
+				poolId:         validPoolId,
+				lowerTick:      -50,
+				upperTick:      50,
+				liquidityDelta: DefaultLiquidityAmt,
+			},
+			positionExists:    true,
+			expectedLiquidity: DefaultLiquidityAmt.Add(DefaultLiquidityAmt),
+		},
+		{
+			name: "Init position for non-existing pool",
+			param: param{
+				poolId:         2,
+				lowerTick:      -50,
+				upperTick:      50,
+				liquidityDelta: DefaultLiquidityAmt,
+			},
+			positionExists: false,
+			expectedErr:    types.PoolNotFoundError{PoolId: 2},
+		},
+		{
+			name: "Init position from -50 to 50 with -50000000000 liquidity",
+			param: param{
+				poolId:         validPoolId,
+				lowerTick:      -50,
+				upperTick:      50,
+				liquidityDelta: DefaultLiquidityAmt.Neg(),
+			},
+			positionExists: false,
+			expectedErr:    types.NegativeLiquidityError{Liquidity: DefaultLiquidityAmt.Neg()},
+		},
+	}
+
+	for _, test := range tests {
+		s.Run(test.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a CL pool with poolId 1
+			_, err := s.App.ConcentratedLiquidityKeeper.CreateNewConcentratedLiquidityPool(s.Ctx, 1, ETH, USDC, DefaultCurrSqrtPrice, DefaultCurrTick)
+			s.Require().NoError(err)
+
+			// If positionExists set, initialize the specified position with defaultLiquidityAmt
+			preexistingLiquidity := sdk.ZeroDec()
+			if test.positionExists {
+				err := s.App.ConcentratedLiquidityKeeper.InitOrUpdatePosition(s.Ctx, test.param.poolId, s.TestAccs[0], test.param.lowerTick, test.param.upperTick, test.param.liquidityDelta)
+				s.Require().NoError(err)
+				preexistingLiquidity = DefaultLiquidityAmt
+			}
+
+			// Get the position info for poolId 1
+			positionInfo, err := s.App.ConcentratedLiquidityKeeper.GetPosition(s.Ctx, 1, s.TestAccs[0], test.param.lowerTick, test.param.upperTick)
+			if test.positionExists {
+				// If we had a position before, ensure the position info displays proper liquidity
+				s.Require().NoError(err)
+				s.Require().Equal(preexistingLiquidity, positionInfo.Liquidity)
+			} else {
+				// If we did not have a position before, ensure getting the non-existent position returns an error
+				s.Require().Error(err)
+				s.Require().ErrorContains(err, types.PositionNotFoundError{PoolId: 1, LowerTick: test.param.lowerTick, UpperTick: test.param.upperTick}.Error())
+			}
+
+			// Initialize or update the position according to the test case
```

Comment:
Just to be clear that here's the main method we are testing since so many other calls are happening in the test case

```suggestion
			// System under test. Initialize or update the position according to the test case
```